### DISCUSSION TITLE
What about to add QPIWS?

### DISCUSSION THREAD

**Discussion by zholdak** [softwarecrash/Solar2MQTT-ESP8266#138] [Ideas]
Hi. 
First of all -- you're did great job, thank you!
What about to add at least raw answer for QPIWS command? It will be great to have warning/fault flags for remote monitoring and automations.
Yes, I know that I can get it by sending custom commands and get an answer, but it isn't convenient...

**Comment by softwarecrash** [comment]
Hello,
The warning and Error Reports are so different by devices that to build in this a verry Big list. Thats the reason why its Not implemented.

**Reply by zholdak** [reply]
Yeap, I understand that. That because I ask for implementation that will query for 'QPIWS' and simple publish an answer into the separate topic... Just that, nothing more for now...
With that simple improvement everyone desired will be able to use it. This may help to accumulate the knowledge base for the future...

As for example -- with my situation not all inverters fully compatible with the firmware and not all device/live data are visible. But I can use the raw data to do it by myself for the concrete inverter.

**Comment by zholdak** [comment]
Hi. I have added a simple RAW QPIWS querying and publishing it to MQTT  (see attached diff [RAW-QPIWS-to-1.1.8.patch](https://github.com/user-attachments/files/17267094/RAW-QPIWS-to-1.1.8.patch)). It works as expected, so, I will use it for my needs. 
You can use it and add into the future releases, if you wish )
![image](https://github.com/user-attachments/assets/134f84ca-b150-4280-932c-5fcd5ffd0bdb)

**Reply by softwarecrash** [reply]
hey, you can make a PR when you tested it, i will merge it into main

**Reply by zholdak** [reply]
I did it. Please, take a look )

**Reply by softwarecrash** [reply]
thank you, its now in the pre release

**Reply by softwarecrash** [reply]
> I did it. Please, take a look )

Short question,
I cant Test it Here, so why is vector used?
Is the Output a "," seperated Error list?
